src/devices/generic_rc_power_socket.c

```diff
@@ -0,0 +1,162 @@
+        return 0;
+
+    // invert the bits, short pulse is 0, long pulse is 1
+    const uint32_t packet = ~(bb[0] << 16 | bb[1] << 8 | bb[2]);
+
+    // Seems OKish up to now, so start decoding the packet and filter out eventual broken packets. These could be broken due to RF interference,
+    // multiple button pressed simultaneosly by the user, or due to improper button debouncing in this oversimplified electrical design.
+
+    // decode the On(1) / Off(0) command
+    uint8_t cmd = 0;
+    switch (packet & 0x0F) {
+        case 0x04: cmd = 1; break;   // on
+
+        case 0x01: break;            // off
+
+        default: 
+            // both command bits or no command bits were set, invalid packet, possibly because both on/off 
+            // buttons were pressed together or because of lack of button debouncing
+          return 0;
+    }
+
+    // decode the remote id (any 5 bit wide value is valid, including 0, leading to 32 unique ids)
+    uint8_t remoteId = 0;
+    for (int i = 0; i < 5; i++) {
+        switch ((packet >> (14 +  i * 2)) & 0x03) {
+            case 0x01: break;                    // floating in remote id field means DIP switch set to 0
+
+            case 0x03: remoteId |= (1 << i); break;    // 1 means DIP switch set to 1
+
+            default: return 0;                   // found an illegal bit combination, give up
+        }
+    }
+
+    // decode the socket id (button name): A to E, represented as a number 1 to 5; exactly one button must be pressed
+    uint8_t socketId = 0;
+    for (int i = 0; i < 5; i++) {
+        switch ((packet >> (4 + i * 2)) & 0x03) {
+            case 0x00: {                          // a transmitted value of 0 actually means a socket id DIP switch set to 1
+                if (socketId != 0) 
+                    return 0;                     // found multiple buttons pressed toghether, invalid packet, give up
+                socketId = 5 - i; 
+                break; 
+            }
+
+            case 0x01: break;                     // floating in socket id field means DIP switch set to 0
+
+            default: return 0;                    // found an illegal bit combination, give up
+        }
+    }
+
+    if (socketId == 0)
+        return 0;  // it is an error to not have any button flag set, give up
+
+    // output tri-state for debugging purposes
+    char tristate[13];
+    char *p = tristate;
+    for (int i = 22; i >= 0; i -= 2) {
+        switch ((packet >> i) & 0x03) {
+            case 0x00: *p++ = '0'; break;
+
+            case 0x01: *p++ = 'Z'; break; // floating / "open"
+
+            case 0x03: *p++ = '1'; break;
+
+            default: return 0;            // found an illegal bit combination, give up
+        }
+    }
+    *p = '\0';
+
+    data_t *data = data_make(
+            "model",        "",             DATA_STRING, "Generic Remote Controlled Power Socket (PT2262/PT2272)",
+            "remoteId",     "Remote Id",    DATA_INT,    remoteId,
```

Comment:
Should all numbers sent to data_make be declared ```int``` ? ```remoteId```, ```socketId```, ```cmd``` are not int and it seems to work, but not sure...